glib/src/object.rs

```diff
@@ -1760,8 +1743,9 @@ impl<T: ObjectType> ObjectExt for T {
         }
 
         // This is actually G_SIGNAL_TYPE_STATIC_SCOPE
-        let return_type: Type =
-            from_glib(details.return_type & (!gobject_ffi::G_TYPE_FLAG_RESERVED_ID_BIT));
+        let return_type: Type = from_glib(
+            signal_query.return_type().to_glib() & (!gobject_ffi::G_TYPE_FLAG_RESERVED_ID_BIT),
```

Comment:
This is actually a problem and should probably already be done as part of the `GSignalQuery` bindings. The "return type" should have that flag removed already or otherwise it's not actually a `GType`, but instead we should have another bool for `fn static_scope(&self) -> bool` that only matches that flag.

Aaaalso: we need to be able to set the flag in the signal builder :)

Reply:
Also the flag can be in all the parameter types of `GSignalQuery` and needs to be removed and handled the same way there...

Reply:
Or maybe for now we can simply unset that flag in `Type::from_glib()`, and handle the static scope stuff in a separate PR later.

Reply:
Aha, interesting. I ended up always removing the extra bit in Type::from_glib but I don't know how to check if it's set in a GType for `static_scope`. For the signal builder part, the extra bit has to be set only in the return_type I assume? or in all the various types?

Reply:
> but I don't know how to check if it's set in a GType for static_scope

What do you mean with that? You can check if it's set by doing `(type_ & gobject_ffi::G_TYPE_FLAG_RESERVED_ID_BIT) != 0` for example.

> For the signal builder part, the extra bit has to be set only in the return_type I assume? or in all the various types?

Return and parameter types.

Reply:
I have pushed something that should handle the static scope stuff :) not very confident about some parts of it ^^

Reply:
Looks good except for the detail that it's a flag per return type / parameter, not one flag for all of them per signal. They can be marked individually as static scope.

Reply:
We can move that to a separate PR if you prefer

Reply:
> We can move that to a separate PR if you prefer

It should be fine :-) unless we can't figure out a nice way to set the static_scope per param/return type, for now I went with (Type, bool) which isn't super clear...